### PR TITLE
Add list functions and properties

### PR DESCRIPTION
**Pull request by limemloh**
Hi there
Here are some definitions and properties from a case study I did together with @paldepind.

This will add the `length` function and `∷ʳ`(snoc) together with related properties and some properties regarding nil.

### CODE REVIEW

File: Cubical/Data/List/Properties.agda [hunk 1]
```diff
@@ -122,6 +122,25 @@ cons-inj₂ = cong safe-tail
 ¬nil≡cons : ∀ {x : A} {xs} → ¬ ([] ≡ x ∷ xs)
 ¬nil≡cons {A = A} p = lower (subst (caseList (List A) (Lift ⊥)) p [])
 
+¬snoc≡nil : ∀ {x : A} {xs} → ¬ (xs ∷ʳ x ≡ [])
+¬snoc≡nil {xs = []} contra = ¬cons≡nil contra
+¬snoc≡nil {xs = x ∷ xs} contra = ¬cons≡nil contra
+
+¬nil≡snoc : ∀ {x : A} {xs} → ¬ ([] ≡ xs ∷ʳ x)
```
Comment (mortberg) [inline Cubical/Data/List/Properties.agda]: Can't you just prove this in one line using `sym` and `¬snoc≡nil`?

Comment (limemloh) [reply Cubical/Data/List/Properties.agda]: Yes, you are right! This would be a better solution.

**Comment by limemloh** [comment]
@mortberg I have adjusted the code to your comment

**Comment by mortberg** [comment]
Thanks for the contribution!

**Pull request merged**